src/main/java/seedu/address/model/id/IdManager.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.model.id;
+
+import seedu.address.model.person.Customer;
+import seedu.address.model.person.Driver;
+import seedu.address.model.task.Task;
+
+/**
+ * Keeps a record of the last unique id used by {@link Task}, {@link Customer} and {@link Driver}.
+ */
+public class IdManager {
```

Comment:
Then expose a public method that will not change id. Abstraction and encapsulation in work here.

Reply:
currently i have lastTaskIdPlusOne to increase the id

Reply:
put all the id increment under model manager. didn't combine the increment and get method because when a new Task is created, it doesn't mean that it will successfully added into the list. It still has to go through AddTaskCommand checks.